recognition/include/pcl/recognition/face_detection/face_common.h

```diff
@@ -117,25 +117,30 @@ namespace pcl
             for (std::size_t j = 0; j < 3; j++)
               stream.write (reinterpret_cast<const char*> (&covariance_rot_ (i, j)), sizeof(covariance_rot_ (i, j)));
 
-          for (int sub_node_index = 0; sub_node_index < num_of_sub_nodes; ++sub_node_index)
+          for (auto sub_node_index : sub_nodes)
           {
-            sub_nodes[sub_node_index].serialize (stream);
+            sub_node_index.serialize (stream);
```

Comment:
Use `auto&` or `const auto&` depending on context (L120, L165). There's no need for making extra copies.